.github/workflows/test_rule_editor_preview.yml

```diff
@@ -5,29 +5,37 @@ on:
   workflow_dispatch:
 
 jobs:
+  rule_tester:
+    uses: ./.github/workflows/rule-tester.yml
+    secrets: inherit
+    with:
+      environment: DEV
+      branch: ${{ github.head_ref || github.ref_name }}
   extract_and_test:
+    needs: rule_tester
```

Comment:
This should be added to automated-CI instead of here. Better modularity and less complexity